source/rust_verify/example/integer_ring/integer_ring.rs

```diff
@@ -116,6 +117,114 @@ pub proof fn multiple_offsed_mod_gt_0(a: nat, b: nat, c: nat) by(nonlinear_arith
     multiple_offsed_mod_gt_0_int(a as int,b as int,c as int, (a%c) as int, (b%c) as int, ((a-b)%(c as int)) as int);
 }
 
+// currently can't use Singular for this proof
+// however, I think we can extend the encoding for div
+pub proof fn FundamentalDivMod(x: int, d: int) by(nonlinear_arith) 
+    requires
+        d > 0,
+    ensures
+        x == d * (x / d) + (x % d)
+{}
+
+// can't use Singular for this proof because of bound
+// however, there might be a way to encode this as an "axiom" in Singular
+// add an equality to the ring whenever we can prove a bound from Z3
+pub proof fn LemmaSmallMod(x: int, m: int) by(nonlinear_arith)
+    requires
+        x >= 0,
+        m > 0,
+        x < m,
+    ensures
+        x % m == x
+{}
+
+// can't use Singular for this proof because of bound
```

Comment:
there are multiple ensures - which one doesn't work?

Reply:
> there are multiple ensures - which one doesn't work?

Then encoding for Singular currently does not allow requires/ensures with inequations such as `0 <= x % m`.